### PR TITLE
HAAR-843: Search for Verified email address for users

### PR DESCRIPTION
**Pull request by ravmoj**
Search for Verified email address for users

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt [hunk 1]
```diff
@@ -328,6 +331,53 @@ class UserController(
...
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Unable to enable user, the user is not within one of your groups.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt:368]: Don't think this applies as not using auth_group_manager role

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt [hunk 2]
```diff
@@ -328,6 +331,53 @@ class UserController(
     val groups = userGroupService.getMyAssignableGroups()
     return groups.map { UserGroupDto(it) }
   }
+
+  @PostMapping("/email")
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt:335]: Should this be a POST request?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt [hunk 3]
```diff
@@ -328,6 +331,53 @@ class UserController(
...
+    ]
+  )
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_ACCESS_ROLES', 'ROLE_MAINTAIN_ACCESS_ROLES_ADMIN')")
+  suspend fun getUsersEmails(
+    @Parameter(description = "List of usernames.", required = true) @RequestBody
+    usernames: List<String>
+  ) =
+    userSearchService
+      .findUsersByUsernames(usernames)
+      .filter { it.verified }
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt:379]: Can't we do this filter in the database? Or if not would be better in the service layer as feels like business logic that the controller should not be doing?

**Review by AlTurner-MOJ — COMMENTED** [review]
As discussed, this end point is obsolete in Auth and therefore can be removed

**Comment by ravmoj** [comment]
Deleting "POST /api/authuser/email" endpoint from Auth. Hence not moving to external users api

**Pull request closed without merging**